Some 1.16 content adaptations

VT-14 opened the pull request:
Note: I am relatively new to modding.  Please verify the code's quality before pulling.

## Added
* Netherite Helmet with Probe (including Knockback Resistance)

## Changed
* Iron, Gold, and Diamond Helmet with Probe textures to match the new 1.14+ base textures.  If 1.14 and/or 1.15 are still being maintained then you may want to backport these textures.
* Renamed "Gold Helmet with Probe" to "Golden Helmet with Probe" in en_us to match vanilla's name.
* TopArmorMaterial now accepts Knockback Resistance values.

## Further Suggestions
#### I'm not sure how to currently, but could look into it some more
* Add Golden Helmet with Probe to makePiglinsNeutral list.  In vanilla, wearing any piece of Gold equipment will make Piglins Neutral.  Additional Notes in "Renamed Gold Helmet Probe" commit.
* Use the Curios API to port the Probe Goggles.
#### I could change easily since it's basically just Data Pack stuff.
* Change the Helmet + Probe = Helmet with Probe recipes to use the vanilla Smithing Table.
**Pro:** Automatically preserves Enchantments and Durability
**Con:** JEI currently does not have a Smithing Table GUI (and vanilla does not have a Recipe Book tab for the Smithing Table either), so it may confuse players.
* Add recipe advancements for items so players get the Toast message, and the recipes appear in the Recipe Book.

VT-14 committed:
Update Vanilla Helmet Probe Textures

Consider Backporting to 1.14 and/or 1.15 versions.

Updated the Helmet textures to use the "new" (1.14+) vanilla armor base textures.

I also ran these textures through TinyPNG.com to reduce their file size.

VT-14 committed:
Add Netherite Helmet with Probe.

Added Netherite Helmet with Probe. Includes 2 Recipes:
1) Netherite Helmet + Probe In Crafting Table.
2) Diamond Helmet w/ Probe + Netherite Ingot in Smithing Table.

Added Knockback Resistance to TopArmorMaterial to match the Netherite Helmet's Knockback Resistance.

VT-14 committed:
Renamed Gold Helmet Probe

Renamed the "Gold Helmet with Probe" to "Golden Helmet with Probe" in en_us.json.  This matches vanilla's name for Golden Armor.

Note: I would really like to make Piglins treat the Golden Helmet with Probe as a Golden Helmet (makes them Neutral).  At this time I am not sure how to do that safely.

At least with the version of Forge currently used (32.0.57), the IForgeItem.class's "makesPiglinsNeutral" method currently checks to see if the armor's material is Gold (== ArmorMaterial.GOLD).

pupnewfster commented on src/main/java/mcjty/theoneprobe/items/TopArmorMaterial.java:
```diff
@@ -13,17 +13,19 @@
     private final int[] damageReduction;
     private final int enchantability;
     private final float toughness;
+    private final float field_234660_o_;
```
This and the parameter can just be called `knockbackResistance` or something rather than copying the srg name that shows for vanilla of `field_234660_o_`

pupnewfster commented on src/main/java/mcjty/theoneprobe/items/TopArmorMaterial.java:
```diff
@@ -64,6 +66,6 @@ public float getToughness() {
     @Override
     public float func_230304_f_() {
-        return 0;
+    	return this.field_234660_o_;
```
The four spaces before the return seem to have been changed into a tab.

VT-14 committed:
Renamed Knockback Resistance variable.  Corrected Spacing.

VT-14 commented:
I've changed `field_234660_o_` to `knockbackResistance`, and corrected that spacing/tab error.

However, I also found that the durability for the helmets are significantly lower than vanilla.  I should have that fixed soon.

VT-14 committed:
Corrected Helmet Durability Values

They were off by a factor of 11.  I included the entire durability multiplier array (`MAX_DAMAGE_ARRAY`) just in case.

VT-14 committed:
Corrected spacing/tab (again).

McJty commented:
Is there a way for me to only merge the content changes? TopArmorMaterial but I don't want that anyway. Just the texture fixes

VT-14 commented:
I'm not super familiar with Git yet, so it would probably be easier for me to just open a new pull requests with just the texture changes.  I'll get that ready tomorrow, or you are welcome to grab anything already on my fork if you want it sooner.

Do you want the Netherite texture, the "Gold" -> "Golden" language file change, or any of the other tweaks mentioned?

McJty commented:
I'm fine with all cosmetic changes that make TOP more in line with modern MC. That includes textures, names, ...
Thanks

VT-14 commented:
With the requested assets changes now submitted separately in #465, I'm going to go ahead and close this PR.
